content/articles/functional-programming-with-java-lambda/index.md

```diff
@@ -0,0 +1,333 @@
+
+#### Lambda example with a single parameter
+
+```java
+
+@FunctionalInterface
+
+interface MyFunctionalInterface {
+
+ //A method with a single parameter
+
+ int incrementByFive(int a);
+
+}
+
+public class LambdaSingleParam {
+
+ public static void main(String args[]) {
+
+ // lambda expression with single parameter num
+
+ MyFunctionalInterface f = (num) -> num+5;
+
+ System.out.println(f.incrementByFive(22));
+
+ }
+
+}
+
+```
+
+#### Lambda example with many parameters
+
+```java
+
+interface StringConcat {
+
+ String strConcat(String a, String b);
+
+}
+
+public class LambdaMultipleParameter {
+
+ public static void main(String args[]) {
+
+ // lambda expression with many arguments
+
+ StringConcat s = (str1, str2) -> str1 + str2;
+
+ System.out.println("Result: "+ s.sconcat("Hello ", "World"));
+
+ }
+
+}
+
+```
+
+### Lambda stream pipeline
+Streams are objects of classes that implement interface Stream. the stream is either from or one of the specific streams interfaces for preparing collections of int, long or double values. Stream enables you to act on the collection of elements with lambda.
+
+Lambda Stream pipeline is a sequence of processing steps from the data source (from collections). The pipeline performs two main operations which is an intermediate or terminal operation. Which formed a chain method calls.
+
+#### Intermediate and terminal Operation
+An intermediate operation is an activity that plays out a particular task on a stream element and the result of the task form a new stream. Intermediate operations are also called Lazy operations. Which implies that the operation isn't performed until a terminal operation is called.
+
+| Operation  |                           Description                           |
+| ---------- | :-------------------------------------------------------------: |
+| filter     | The filter method allows us to get elements of the stream that  |
+|            |                     meet certain conditions.                     |
+| ---------- | :-------------------------------------------------------------: |
+| distinct   |   The distinct method returns distinct elements of the stream.    |
+| ---------- | :-------------------------------------------------------------: |
+| limit      |  The limit method return elements of the stream with the given  |
+|            |                         number or limit.                        |
+| ---------- | :-------------------------------------------------------------: |
+| map        |  The filter method allows us to get elements of the stream that  |
```

Comment:
The description here is the same as the one in the filter operation. Check on that.